File: template/src/main/java/ru/surfstudio/standard/ui/util/URLSpanNoUnderline.java
package ru.surfstudio.standard.ui.util;


import android.annotation.SuppressLint;
import android.text.TextPaint;
import android.text.style.URLSpan;

@SuppressLint("ParcelCreator")
public class URLSpanNoUnderline extends URLSpan {
    public URLSpanNoUnderline(String url) {
        super(url);
    }

    public void updateDrawState(TextPaint textPaint) {
        super.updateDrawState(textPaint);
        textPaint.setUnderlineText(false);
    }
}

File: util-ktx/src/main/java/ru/surfstudio/android/util/URLSpanNoUnderline.kt
package ru.surfstudio.android.util


import android.annotation.SuppressLint
import android.text.TextPaint
import android.text.style.URLSpan

@SuppressLint("ParcelCreator")
class URLSpanNoUnderline(url: String) : URLSpan(url) {

    override fun updateDrawState(textPaint: TextPaint) {
        super.updateDrawState(textPaint)
        textPaint.isUnderlineText = false
    }
}
